ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -554,6 +554,11 @@ func (r *RayJobReconciler) getOrCreateRayClusterInstance(ctx context.Context, ra
 	}
 	r.Log.Info("Found associated RayCluster for RayJob", "RayJob", rayJobInstance.Name, "RayCluster", rayClusterNamespacedName)
 
+	if len(rayJobInstance.Spec.ClusterSelector) == 0 && !utils.CompareJsonStruct(rayClusterInstance.Spec, *rayJobInstance.Spec.RayClusterSpec) {
```

Comment:
Add comments for why do we need to add `len(rayJobInstance.Spec.ClusterSelector) == 0`.